### PR TITLE
v0 of templated service code

### PR DESCRIPTION
**Pull request by d00rman**
This is just v0. A lot of things are still missing, but let's start somewhere :)

### CODE REVIEW

File: routes/v1.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ */
+var app;
+
+
+/**
+ * GET /conf
+ * Reads and returns the configuration file used to
+ * start the service. IMPORTANT NOTE: This is a sample
+ * only for demostrating promises, you should not
+ * expose this endpoint in real services.
```
Comment (gwicke) [inline routes/v1.js]: Maybe it's better not to include unsafe examples. That config might contain private info like passwords.

Perhaps provide something like fibonacci instead?

**Comment by gwicke** [comment]
I wonder if https://github.com/krakenjs/swaggerize-express could be useful for our purposes.

File: app.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+
+var express = require('express');
+var fs = Promise.promisifyAll(require('fs'));
```
Comment (gwicke) [inline app.js]: Better to use a local Promise implementation here. The global es6 Promise doesn't actually have `promisifyAll`.

Example:

``` javascript
var P = require('bluebird');
var fs = P.promisifyAll(require('fs'));
```

File: routes/v1.js (hunk 1 above)
Comment (d00rman) [inline routes/v1.js]: Definitely. I put it there more as a place holder than the definitive version. Will address that soon (either in this or another PR)

File: app.js (hunk 2 above)
Comment (d00rman) [inline app.js]: Actually, I was relying on `service-runner`'s globalisation of bluebird, but now that that's gone, I'll change the code accordingly.

File: config.dev.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+#  streams:
+#  # Use gelf-stream -> logstash
+#  - type: gelf
+#    host: logstash1003.eqiad.wmnet
+#    port: 12201
+
+# Statsd metrics reporter
+metrics:
+  type: txstatsd
+  host: localhost:8125
```
Comment (gwicke) [inline config.dev.yaml]: I moved the port to a separate port: property.

File: config.prod.yaml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  streams:
+  # Use gelf-stream -> logstash
+  - type: gelf
+    host: logstash1003.eqiad.wmnet
+    port: 12201
+
+# Statsd metrics reporter
+metrics:
+  type: txstatsd
+  host: localhost:8125
```
Comment (gwicke) [inline config.prod.yaml]: Same here (separate port attribute).

File: config.dev.yaml (hunk 3 above)
Comment (d00rman) [inline config.dev.yaml]: Ah ok. Will change. (Idem for the comment below)

**Comment by d00rman** [comment]
Current status:
- the simple template infrastructure is in place
- there are some routes showing
  - service information retrieval
  - simple responses
  - dealing with named and optional URI parameters
  - the usage of bluebird and preq
  - test scripts and coverage

Still to do:
- DOM parsing
- logging && metrics
- proper error handling
- documentation

**Pull request merged**